client/components/domains/register-domain-step/index.jsx

```diff
@@ -764,11 +764,7 @@ class RegisterDomainStep extends React.Component {
 
 		const cleanedQuery = getDomainSuggestionSearch( searchQuery, MIN_QUERY_LENGTH );
 		const loadingResults = Boolean( cleanedQuery );
-		const isInitialQueryActive = searchQuery === this.props.suggestion;
-
-		if ( isEmpty( cleanedQuery ) ) {
-			return;
-		}
```

Comment:
Choosing to continue to update the state at this point (no valid query) is going to clear the results indeed. Unfortunately, this will undo what @klimeryk was trying to improve in https://github.com/Automattic/wp-calypso/pull/39664
So I've updated the A/C in our new issue https://github.com/Automattic/wp-calypso/issues/53752:

> - [ ] When the search query is cleared in any way, show empty state for the domain results; 

to 

> - [ ] When the search query is cleared (or reduced to 1 character), show the previous/default domain results;

Reply:
With https://github.com/Automattic/wp-calypso/pull/54124/commits/57e758ff5b3e11188369b60ac9c7219c4f090f2d we don't reset the suggestions when the search query is changed. The loading state will make sure they remain hidden anyway. 

Fixed behavior looks like this:
https://user-images.githubusercontent.com/14192054/124305375-af115900-db6d-11eb-99c6-b7d71bd966a9.mov

* No results => clearing the search query while suggestions loading renders empty state
* Existing results => clearing renders previous results